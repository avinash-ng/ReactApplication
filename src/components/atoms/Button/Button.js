import React from 'react'
import PropTypes from 'prop-types'
const Button = (props ) => {
    return (
        <button className="a_button" className={props.className} onClick={props.onClicked}>{props.buttonName}</button>
    )
}

Button.propTypes = {
    buttonName: PropTypes.string.isRequired
}

export default Button
