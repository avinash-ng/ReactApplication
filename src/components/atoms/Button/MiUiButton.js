import React from 'react'
import { Button } from "@material-ui/core"
const MiUiButton = props => {
    return (
        <Button className={props.className} style={props.style} variant={props.variant} onClick={props.onClick}>{props.buttonName}</Button> 
    )
}
export default MiUiButton