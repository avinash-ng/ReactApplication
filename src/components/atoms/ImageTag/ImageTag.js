import React from 'react'
import './ImageTag.css'
const ImageTag = props => {
    return (
        <h3 className={props.className} style={{ width: '450px', marginBottom: '20px' }} onClick={props.onClicked} >{props.object.name}</h3>
    )
}
export default ImageTag