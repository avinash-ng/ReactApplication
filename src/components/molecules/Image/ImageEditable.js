import React from 'react'
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from "../../atoms/Button/Button"
const ImageEditable = props => {
    return (
        <div className="tag-div" contentEditable="true" suppressContentEditableWarning={true}>
            {/* <h3 className = "h3-image-tag" style={{ width: '450px' }}  onClick={() => changeHeading(each, index)} >{each.name}</h3> */}
            <ImageTag className={props.className} onClicked={props.headingOnClicked} object={props.object}></ImageTag>
            {props.object.selected ? <Button buttonName={props.buttonName} onClicked={props.buttonOnClicked}></Button> : ""}

        </div>
    )
}
export default ImageEditable