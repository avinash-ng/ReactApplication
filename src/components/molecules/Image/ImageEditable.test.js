import ImageEditable from './ImageEditable'
import expect from 'expect'
import sinon from 'sinon';
import { shallow, configure, mount } from 'enzyme'
import React from 'react'
import Adapter from 'enzyme-adapter-react-16';
import { render } from '@testing-library/react';
import ImageTag from '../../atoms/ImageTag/ImageTag'
import Button from '../../atoms/Button/Button'

configure({ adapter: new Adapter() });
describe('<ImageEditable /> ', () => {
    it('renders <ImageTag /> element properly', () => {
        const object = {
            'selected': 'false'
        }
        const buttonName='click me'
        const RenderComponent = shallow(
            <ImageEditable object={object} buttonName={buttonName}></ImageEditable>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(ImageTag).exists()).toBe(true)
    })
    
    it('renders <Button /> element properly', () => {
        const object = {
            'selected': 'true'
        }
        const buttonName='click me'
        const RenderComponent = shallow(
            <ImageEditable object={object} buttonName={buttonName}></ImageEditable>
        );
        // console.log(RenderComponent)
        expect(RenderComponent.find(Button).exists()).toBe(true)
    })
    it('recieved props correctly and check on click functionality', () => {
        const object = {
            'selected': true
        }
        const className = 'image';
        const buttonName = 'click me';
        const buttonOnClicked = sinon.spy();
        const headingOnClicked = sinon.spy();
        const RenderComponent = mount(<ImageEditable buttonName={buttonName} buttonOnClicked={buttonOnClicked} headingOnClicked={headingOnClicked} object={object} className={className} ></ImageEditable>)
        RenderComponent.find(Button).simulate('click')
        expect(buttonOnClicked).toHaveProperty('callCount', 1)
    })


    it('Matches snapshot', () => {
        const object = {
            'selected': false
        }
        const className = 'image';
        const buttonName = 'click me';
        const buttononClicked = sinon.spy();
        const headingOnClicked = sinon.spy();
        const RenderComponent = mount(<ImageEditable buttonName={buttonName} buttononClicked={buttononClicked} headingOnClicked={headingOnClicked} object={object} className={className} ></ImageEditable>)
        expect(RenderComponent).toMatchSnapshot()
    })
})