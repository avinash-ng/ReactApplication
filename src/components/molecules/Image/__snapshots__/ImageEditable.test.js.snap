// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageEditable />  Matches snapshot 1`] = `ReactWrapper {}`;
